### PR TITLE
Add test cases for API utilities

### PR DESCRIPTION
**Pull request by mzfr**
I've added some unit tests for [utility.py](https://github.com/zdresearch/OWASP-Honeypot/blob/master/api/utility.py) in API

### CODE REVIEW

**Comment by dhirensr** [comment]
@mzfr : Thank you for the PR :smile:  
@Ali-Razmjoo : this PR is adding tests but maybe we should remove the tests from utils and create a pytest environment for the project so that we can add tests for new functions later on. what do you think?

**Comment by Ali-Razmjoo** [comment]
I liked the idea, let's improve the PR and make a whole test for our CI?

**Comment by dhirensr** [comment]
@Ali-Razmjoo : we need to install pytest via travis to check if tests pass and also in the requirements file. pytest will always run the files starting with test_*.py .
also mentioning about tests in wiki would be good for other contributors! 
@mzfr :  I have reviewed the PR and also I think file name should be changed to test_api_utility.py ,file names with - are not pythonic way of writing and also pytest would need test_*.py format.

**Comment by mzfr** [comment]
It would be nice if we can have `requirements-dev.txt` in which we can place `pytest` so the normal user won't have to install pytest and only people who are going to develop the tool will install it.

**Comment by dhirensr** [comment]
@mzfr : that would be a good idea! i would make the environment changes with travis-ci ready and then we can check.

**Comment by Ali-Razmjoo** [comment]
@dhirensr merged #57, you can start enhancing this PR.

**Comment by dhirensr** [comment]
@Ali-Razmjoo : there is a bug in travis-ci, it's not actually running pytest in the script and hence no tests are running.

**Comment by dhirensr** [comment]
@mzfr : did you check if the tests are running locally with pytest? for me it's failing,please make the changes so that it passes.
I am getting 
```
E   ModuleNotFoundError: No module named 'api'
```

**Comment by mzfr** [comment]
How are you running those tests and from which directory?

**Comment by dhirensr** [comment]
@mzfr :
trree

**Comment by Ali-Razmjoo** [comment]
I've merged @dhirensr's pr, can you rebase to master branch?

File: tests/test_api_utilities.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+    def test_fix_date_3(self):
+        returned_values = fix_date("02/05/2019 | 01/05/2019")
+        expected_output = ['02/05/2019  00:00:00', ' 01/05/2019 23:59:59']
+        self.assertListEqual(expected_output, returned_values)
+
+    def test_fix_date_4(self):
+        returned_values = fix_date("02/05/2019")
+        expected_output = ['02/05/2019 00:00:00', '02/05/2019 23:59:59']
+        self.assertListEqual(expected_output, returned_values)
```
Comment (Ali-Razmjoo) [inline tests/test_api_utilities.py]:
```suggestion
import unittest

from api.utility import fix_date
from api.utility import fix_limit
from api.utility import fix_skip
from api.utility import msg_structure


class TestApiUtility(unittest.TestCase):

    def test_msg_structure_pass(self):
        self.assertEqual(
            msg_structure("ok", "Hello OWASP"),
            {
                "status": "ok",
                "msg": "Hello OWASP"
            }
        )

    def test_msg_structure_fail(self):
        self.assertNotEqual(
            msg_structure("failed", "Hello again"),
            {
                "status": "ok",
                "msg": "Hello again"
            }
        )

    def test_fix_skip_pass(self):
        returned_values = fix_skip('0')
        self.assertTrue(isinstance(returned_values, int))
        self.assertEqual(returned_values, 0)

    def test_fix_skip_fail(self):
        returned_values = fix_skip('0')
        self.assertFalse(isinstance(returned_values, str))
        self.assertNotEqual(returned_values, 1)

    def test_fix_limit_pass(self):
        returned_values = fix_limit('10')
        self.assertEqual(returned_values, 10)
        self.assertTrue(isinstance(returned_values, int))

    def test_fix_limit_fail(self):
        returned_values = fix_limit('0')
        self.assertNotEqual(returned_values, 11)
        self.assertFalse(isinstance(returned_values, str))

    def test_fix_date(self):
        self.assertListEqual(
            fix_date("02:05:2019"),
            ['02:05:2019', '02:05:2019 23:59:59']
        )
        self.assertListEqual(
            fix_date("02:05:2019 | 01:05:2019"),
            ['02:05:2019 ', ' 01:05:2019']
        )
        self.assertListEqual(
            fix_date("02/05/2019 | 01/05/2019"),
            ['02/05/2019  00:00:00', ' 01/05/2019 23:59:59']
        )
        self.assertListEqual(
            fix_date("02/05/2019"),
            ['02/05/2019 00:00:00', '02/05/2019 23:59:59']
        )
```

Comment (Ali-Razmjoo) [inline tests/test_api_utilities.py]:
```suggestion
import unittest

from api.utility import fix_date
from api.utility import fix_limit
from api.utility import fix_skip
from api.utility import msg_structure


class TestApiUtility(unittest.TestCase):

    def test_msg_structure_pass(self):
        self.assertEqual(
            msg_structure("ok", "Hello OWASP"),
            {
                "status": "ok",
                "msg": "Hello OWASP"
            }
        )

    def test_msg_structure_fail(self):
        self.assertNotEqual(
            msg_structure("failed", "Hello again"),
            {
                "status": "ok",
                "msg": "Hello again"
            }
        )

    def test_fix_skip_pass(self):
        returned_values = fix_skip('0')
        self.assertTrue(isinstance(returned_values, int))
        self.assertEqual(returned_values, 0)

    def test_fix_skip_fail(self):
        returned_values = fix_skip('0')
        self.assertFalse(isinstance(returned_values, str))
        self.assertNotEqual(returned_values, 1)

    def test_fix_limit_pass(self):
        returned_values = fix_limit('10')
        self.assertEqual(returned_values, 10)
        self.assertTrue(isinstance(returned_values, int))

    def test_fix_limit_fail(self):
        returned_values = fix_limit('0')
        self.assertNotEqual(returned_values, 11)
        self.assertFalse(isinstance(returned_values, str))

    def test_fix_date(self):
        self.assertListEqual(
            fix_date("02:05:2019"),
            ['02:05:2019', '02:05:2019 23:59:59']
        )
        self.assertListEqual(
            fix_date("02:05:2019 | 01:05:2019"),
            ['02:05:2019 ', ' 01:05:2019']
        )
        self.assertListEqual(
            fix_date("02/05/2019 | 01/05/2019"),
            ['02/05/2019  00:00:00', ' 01/05/2019 23:59:59']
        )
        self.assertListEqual(
            fix_date("02/05/2019"),
            ['02/05/2019 00:00:00', '02/05/2019 23:59:59']
        )

```

**Comment by mzfr** [comment]
@Ali-Razmjoo I've updated the tests in the way you requested. Also now the test cases are compatible with `pytest`.

To check just go in the `test/` directory and run `pytest`.

image

File: tests/test_api_utilities.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import unittest
```
Comment (Ali-Razmjoo) [inline tests/test_api_utilities.py]:
```suggestion
#!/usr/bin/env python
# -*- coding: utf-8 -*-

import unittest
```

**Pull request merged**